clients/ts/signalr/src/HttpClient.ts

```diff
@@ -5,30 +5,89 @@ import { AbortSignal } from "./AbortController";
 import { HttpError, TimeoutError } from "./Errors";
 import { ILogger, LogLevel } from "./ILogger";
 
+/** Represents an HTTP request. */
 export interface HttpRequest {
+    /** The HTTP method to use for the request. */
     method?: string;
+
+    /** The URL for the request. */
     url?: string;
+
+    /** The body content for the request. May be a string or an ArrayBuffer (for binary data). */
     content?: string | ArrayBuffer;
+
+    /** An object describing headers to apply to the request. */
     headers?: { [key: string]: string };
+
+    /** The XMLHttpRequestResponseType to apply to the request. */
     responseType?: XMLHttpRequestResponseType;
+
+    /** An AbortSignal that can be monitored for cancellation. */
     abortSignal?: AbortSignal;
+
+    /** The time to wait for the request to complete before throwing a TimeoutError. Measured in milliseconds. */
     timeout?: number;
 }
 
+/** Represents an HTTP response. */
 export class HttpResponse {
+    /** Constructs a new instance of {@link HttpResponse} with the specified status code.
+     *
+     * @param {number} statusCode - The status code of the response.
```

Comment:
How come these param docs are separated by a dash?

Reply:
It's "optional" but I thought it was required initially. I'll tidy that up.